src/Consolonia.PlatformSupport/CursesConsole.cs

```diff
@@ -608,107 +612,70 @@ private void HandleMouseInput(Curses.MouseEvent ev)
-                        RaiseMouseClickedEvent(RawPointerEventType.LeftButtonDown, RawPointerEventType.LeftButtonUp,
-                            1, point, rawInputModifiers);
-                        break;
-                    case Curses.Event.Button1DoubleClicked:
-                        RaiseMouseClickedEvent(RawPointerEventType.LeftButtonDown, RawPointerEventType.LeftButtonUp,
-                            2, point, rawInputModifiers);
-                        break;
-                    case Curses.Event.Button1TripleClicked:
-                        RaiseMouseClickedEvent(RawPointerEventType.LeftButtonDown, RawPointerEventType.LeftButtonUp,
-                            3, point, rawInputModifiers);
-                        break;
-                    case Curses.Event.Button2Clicked:
-                        RaiseMouseClickedEvent(RawPointerEventType.MiddleButtonDown,
-                            RawPointerEventType.MiddleButtonUp, 1, point, rawInputModifiers);
-                        break;
-                    case Curses.Event.Button2DoubleClicked:
-                        RaiseMouseClickedEvent(RawPointerEventType.MiddleButtonDown,
-                            RawPointerEventType.MiddleButtonUp, 2, point, rawInputModifiers);
-                        break;
-                    case Curses.Event.Button2TripleClicked:
-                        RaiseMouseClickedEvent(RawPointerEventType.MiddleButtonDown,
-                            RawPointerEventType.MiddleButtonUp, 3, point, rawInputModifiers);
-                        break;
-                    case Curses.Event.Button3Clicked:
-                    case Curses.Event.Button4Clicked:
-                        RaiseMouseClickedEvent(RawPointerEventType.RightButtonDown,
-                            RawPointerEventType.RightButtonUp, 1, point, rawInputModifiers);
-                        break;
-                    case Curses.Event.Button3DoubleClicked:
-                    case Curses.Event.Button4DoubleClicked:
-                        RaiseMouseClickedEvent(RawPointerEventType.RightButtonDown,
-                            RawPointerEventType.RightButtonUp, 2, point, rawInputModifiers);
-                        break;
-                    case Curses.Event.Button3TripleClicked:
-                    case Curses.Event.Button4TripleClicked:
-                        RaiseMouseClickedEvent(RawPointerEventType.RightButtonDown,
-                            RawPointerEventType.RightButtonUp, 3, point, rawInputModifiers);
-                        break;
                     case Curses.Event.Button1Pressed:
-                        _moveModifers = rawInputModifiers | RawInputModifiers.LeftMouseButton;
-                        RaiseMouseEvent(RawPointerEventType.LeftButtonDown, point, wheelDelta, _moveModifers);
+                        RaiseMouseEvent(RawPointerEventType.LeftButtonDown, point, null,
+                            rawInputModifiers | RawInputModifiers.LeftMouseButton);
```

Comment:
I asked you this question also. Why do we pass the state one more time into modifiers, should not be they empty i no CTRL or SHIFT etc were pressed?

Reply:
Yes. at least ncurses specify that only 1 bit is set in each event.

No, my question is about why do we set modifiers to LEFTMOUSEBUTTON when we send LeftButtonDown event?

Reply:
I have rewritten this code again. I had a regression because I wasn't tracking _moveModifiers. For drag to work we need to remember modifier flags and merge them into the MOVE event.